check-sources.sh: move file lists to env variables

jibi opened the pull request:
to allow alternative implementations to expand them.

With that in mind, make the awk logic used to match filenames a bit more
generic.

rgo3 commented on pkg/monitor/datapath_drop.go:
```diff
@@ -62,7 +64,6 @@ var sourceFileNames = map[int]string{
 	105: "nodeport.h",
 	106: "lb.h",
 	107: "mcast.h",
-	//end
```
Uber nit since we already touch this line:
Why not use `// @@ source files list end` and match against that instead of `}` in the awk script to have the same matching mechanism in both files?

rgo3 reviewed (approved):
Just a nit, otherwise LGTM.

jibi committed:
check-sources.sh: move file lists to env variables

to allow alternative implementations to expand them.

With that in mind, make the awk logic used to match filenames a bit more
generic.

Signed-off-by: Gilberto Bertin <jibi@cilium.io>
